### PR TITLE
Validate SqoopHook connection string and disable extra options from public hook methods

### PR DESCRIPTION
**Pull request by pankajkoti**
Check that the connection string constructed using the connection's 
`host`, `port` and `schema` does not contain query params as it is 
not intended. Additionally, also disable the `extra_import_options` 
and `extra_export_options` arguments accepted directly by the hook 
methods but accept it as a param via the hook constructor when 
initialising the hook or by passing it in `hook_params` when initialising 
the hook from operators.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Pending tests passing :)

**Review by potiuk — APPROVED** [review]

**Comment by pankajkoti** [comment]
PR is ready for review
cc: @potiuk @eladkal

File: airflow/providers/apache/sqoop/CHANGELOG.rst [hunk 1]
```diff
@@ -28,6 +28,17 @@
 Changelog
 ---------
 
+4.0.0
```
Comment (eladkal) [inline airflow/providers/apache/sqoop/CHANGELOG.rst:31]: please add 4.0.0 entry to provider.yaml

Comment (pankajkoti) [reply airflow/providers/apache/sqoop/CHANGELOG.rst:31]: sorry, missed adding this earlier. Pushed a commit now to reflect this in the provider.yaml

**Comment by pankajkoti** [comment]
Test constraints check seems to be timing out repeatedly after 80 minutes.

**Comment by potiuk** [comment]
>  Test constraints check seems to be timing out repeatedly after 80 minutes.

Yes. Seems there is a backtracking of `pip` in main (I.e. it has difficulties in solving recent constraints) after some dependency release. will take a look.

**Comment by potiuk** [comment]
BTW. I just realised that PR's like this should not cause the "--upgrade-to-newer-dependencies` (this is what triggers the backtracking issue)  fix to CI infrastructure is coming

**Comment by potiuk** [comment]
After this one is merged : https://github.com/apache/airflow/pull/33082 and after your rebase, the constraint issue should go away even if we do not fix it in main.

**Comment by potiuk** [comment]
Merging, regardless.

**Pull request merged**